diff --git a/modules/hid_t2magicmouse/hid_t2magicmouse.c b/modules/hid_t2magicmouse/hid_t2magicmouse.c
--- a/modules/hid_t2magicmouse/hid_t2magicmouse.c
+++ b/modules/hid_t2magicmouse/hid_t2magicmouse.c
@@ -76 +76 @@
-#if LINUX_VERSION_CODE < KERNEL_VERSION(7, 0, 0)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 19, 0)
diff --git a/modules/t2bce_core/t2bce_main.c b/modules/t2bce_core/t2bce_main.c
--- a/modules/t2bce_core/t2bce_main.c
+++ b/modules/t2bce_core/t2bce_main.c
@@ -764 +764 @@
-MODULE_SOFTDEP("post: t2bce_vhci");
+MODULE_SOFTDEP("pre: t2hid hid_t2magicmouse post: t2bce_vhci");
diff --git a/modules/t2touchbar/t2hid.c b/modules/t2touchbar/t2hid.c
--- a/modules/t2touchbar/t2hid.c
+++ b/modules/t2touchbar/t2hid.c
@@ -818 +818 @@
-	rep = kmalloc_obj(*rep);
+	rep = kmalloc(sizeof(*rep), GFP_KERNEL);
